### ISSUE TITLE
Properly implement the TIOCSCTTY ioctl 

### ISSUE THREAD

**Issue by SuchAFuriousDeath** [obhq/obliteration#795]
This is a standard FreeBSD ioctl. I decided to conciously skip it for now, because:
1) The code would be really messy with a lot of .read() and .write() and .unwrap
2) We would have to use Weak pointers to prevent memory leaks, causing even more unwraps
3) It doesn't produce any output or affect any internal state that affects anything else (as far as we know)
4) previously,before the filesystem rework, this was also skipped and nothing bad happened.

Under these circumstances, I believe it's justifiable to postpone the implementation of this until it is required.
We already got stuck on the filesystem for such a long time and I don't want to get stuck on this right now.

**Comment by SuchAFuriousDeath** [comment]
This is not needed anymore after #810

**Comment by ultimaweapon** [comment]
This one is going to be needed when we supports running the PS4 firmware so I'll re-open this.

**Comment by SuchAFuriousDeath** [comment]
When do you intend to begin working on that, if I may ask?

**Comment by ultimaweapon** [comment]
Once I finished migrating the game to run inside a VM.
